src/uct/ib/rdmacm/rdmacm_cm.c

```diff
@@ -43,9 +43,14 @@ ucs_status_t uct_rdmacm_cm_ack_event(struct rdma_cm_event *event)
 
 ucs_status_t uct_rdmacm_cm_reject(struct rdma_cm_id *id)
 {
+    uct_rdmacm_priv_data_hdr_t hdr;
+
+    hdr.length = 0;
+    hdr.status = (uint8_t)UCS_ERR_REJECTED;
```

Comment:
better to define hdr.status as `int8_t` to avoid signed/unsigned casting

Reply:
but UCS_ERR_REJECTED is also a negative value

Reply:
yes, what's the problem if header.status is negative and stored as int8_t?

Reply:
there is no problem.. that's ok, i'll update.

Reply:
@evgeny-leksikov this status field is also used in the previous rdmacm flow (in rdmacm_iface), which imo we should not change at this point.

Reply:
AFAIR in old flow it's used only for UCS_OK check. If so, it's binary compatible.
Also rdmacm_iface is going to be deprecated, so, IMHO it's good point to fix the wire protocol.

Reply:
it's also checked for the private data in the reject message.

Reply:
actually, tcp_sockcm has almost the same struct so maybe we should unite and have this struct in the cm base class.
@yosefe i can do as part of the PR to unite sockaddr tls code, wdyt?

Reply:
yes to unite, and the status field type should be ucs_status_t IMO

Reply:
ok will unite in another PR along with the rest of the code we planned to unite.

Reply:
not addressed

Reply:
@evgeny-leksikov i will change this in another PR as stated above (along with moving this struct to the cm base class)

Reply:
ok